hdwallet/src/main/kotlin/tech/figure/hdwallet/wallet/Wallet.kt

```diff
@@ -130,6 +130,12 @@ interface Account {
      */
     val keyPair: ECKeyPair
 
+    /**
+     * Test if the account is considered a root account. An account is considered a root if it is not been derived
+     * from a parent [Account].
+     */
+    fun isRoot(): Boolean
```

Comment:
@mtps Let me know if have any objections to exposing this. Upstream of this code, I've found it to be useful to know if an account is a root account or a derived child account and fail fast of an account is not a root.

I guess it's probably fine to expose it considering it's easy to determine if an account is a root by repeatedly deriving children from it until an exception is raised, and keeping track of how deep the tree of child accounts is.

Reply:
I agree with you. Exposing it isn't a big deal and can easily be verified anyway. It's not secret information.